API.md

```diff
@@ -531,6 +536,63 @@ onceFn(); // results in undefined
 
 A simple no-op function. It does nothing at all.
 
+### promiseWrap(bind, method, args, transform)
+
+Returns a new promise that is resolved with what the `method` is called with as first argument.
+`transform` is optional , if supplied it will apply `Hoek.transform(arguments, transform)` on the resolved/rejected object.
+It only rejects if first argument of callback is `instanceOf Error`
+
+```javascript
+
+const simpleSuccess = function (callback) {
+
+    if (!callback) {
+        return Hoek.promiseWrap(this, simpleSuccess);
+    }
+
+    callback(null, 'success');
+};
+
+return simpleSuccess().then((result) => {
+
+    // result is "success"
+});
+```
+
+```javascript
+
+const complexSuccess = function (callback) {
+
+    if (!callback) {
+        return Hoek.promiseWrap(this, complexSuccess, null, ['error', 'value1', 'value2']);
+    }
+
+    callback(null, 'success', 'works');
+};
+
+return complexSuccess().then((result) => {
+
+    // result is {error: null, "value1": "success", "value2": "works"}
```

Comment:
Not convinced it should contain an error at all, it should have gone to the .catch if we expected an error.